Userland/Libraries/LibWebView/ProcessManager.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Types.h>
+#include <AK/Vector.h>
+#include <LibCore/EventReceiver.h>
+#include <LibWebView/Forward.h>
+
+#pragma once
+
+namespace WebView {
+
+enum ProcessType {
+    Chrome,
+    WebContent,
+    WebWorker,
+    SQLServer,
+    RequestServer,
+    ImageDecoder,
+};
+
+struct ProcessInfo {
+    ProcessType type;
+    pid_t pid;
+    u64 memory_usage_kib = 0;
+    float cpu_percent = 0.0f;
+};
+
+ProcessType process_type_from_name(StringView);
+StringView process_name_from_type(ProcessType type);
+
+class ProcessManager : public Core::EventReceiver {
```

Comment:
Does it need to be a `Core::EventReceiver`? doesn't seem to use anything from the base class